lib/services/services_linux.c

```diff
@@ -476,8 +477,25 @@ action_launch_child(svc_action_t *op)
     setpgid(0, 0);
 
     // Close all file descriptors except stdin/stdout/stderr
-    for (lpc = getdtablesize() - 1; lpc > STDERR_FILENO; lpc--) {
-        close(lpc);
+    dir = opendir("/proc/self/fd");
+    if (dir == NULL) { /* /proc not available */
```

Comment:
Most non-Linux OSes support /dev/fd for the same purpose (and /dev/fd is usually a symlink to /proc/self/fd on Linux), so I think we should check for one and then the other if the first didn't exist. FYI, Pacemaker has a SUPPORT_PROCFS constant, so you could wrap the /proc portion in #if SUPPORT_PROCFS to skip that on systems that don't have it (not that an extra opendir() is a big deal).